File: 122_best_time_to_buy_and_sell_stock_2/maxProfit.ts
function maxProfit(prices: number[]): number {
  let res = 0;

  for (let i = 1; i < prices.length; i++) {
    if (prices[i] > prices[i - 1]) {
      res += prices[i] - prices[i - 1];
    }
  }

  return res;
}

File: 122_best_time_to_buy_and_sell_stock_2/max_profit.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut res = 0;
        let mut cur = prices.get(0).copied().unwrap();

        for &p in prices.iter() {
            if p > cur {
                res += p - cur;
            }

            cur = p;
        }
        res as i32
    }
}